Zend/zend_attributes.c

```diff
@@ -380,6 +404,10 @@ void zend_register_attribute_ce(void)
 
 	zend_ce_override = register_class_Override();
 	zend_mark_internal_attribute(zend_ce_override);
+
+	zend_ce_deprecated = register_class_Deprecated();
+	attr = zend_mark_internal_attribute(zend_ce_deprecated);
+	attr->validator = validate_deprecated_attribute;
```

Comment:
I think this can just be set to `NULL`.

Reply:
Yes, but I'm not quite sure how to go about handling the parameter validation.

I don't want to unnecessarily disallow `#[\Deprecated(123)]` or `#[\Deprecated(MY_MESSAGE)]` by explicitly checking for a string parameter during compile time. But compile time validation is generally preferable to catch errors more quickly. Depending on the final choice made, the validator function will be removed entirely or restored.

This is also somewhat related to #13970.

Reply:
Handling of attribute arguments is currently inconsistent, as I've discovered when looking at GH-13970. For `#[Attribute(...)]`, the argument is resolved at compile time and arity is ignored, which is inconsistent with other internal functions. `Attribute` is the only internal attribute accepting arguments, so we can't compare it to anything else. I don't know what the "intended" approach is, but generally I believe resolving of the arguments should be delayed to runtime. It doesn't matter much for `Attribute`, because there are pre-defined constants for it to use that are available at compile time, but the same cannot be said for `Deprecated`. E.g. it would fail when using class constants that have not been auto-loaded yet, which would be weird.

Reply:
Now removed, following the lead of the `Attribute` attribute.